packages/sanity/src/core/form/studio/FormBuilderInputErrorBoundary.tsx

```diff
@@ -0,0 +1,80 @@
+import {Box, Card, Code, ErrorBoundary, Stack, Text} from '@sanity/ui'
+import {useCallback, useMemo, useState} from 'react'
+import {useHotModuleReload} from 'use-hot-module-reload'
+
+import {SchemaError} from '../../config'
+import {useTranslation} from '../../i18n'
+import {CorsOriginError} from '../../store'
+import {isRecord} from '../../util'
+import {Alert} from '../components/Alert'
+
+/**
+ * @internal
+ */
+interface FormBuilderInputErrorBoundaryProps {
+  children: React.ReactNode
+}
+
+/**
+ * @internal
+ */
+export function FormBuilderInputErrorBoundary(
+  props: FormBuilderInputErrorBoundaryProps,
+): JSX.Element {
+  const {children} = props
+  const [{error}, setError] = useState<{error: unknown}>({error: null})
+  const handleRetry = useCallback(() => setError({error: null}), [])
+
+  if (!error) {
+    return <ErrorBoundary onCatch={setError}>{children}</ErrorBoundary>
+  }
+
+  return <ErrorCard error={error} onRetry={handleRetry} />
+}
+
+/**
+ * The error UI is in a separate component to keep the wrapper lightweight,
+ * it allows us to use hooks like useTranslation without incurring the cost of it on every form input
+ * when there are no errors.
+ * @internal
+ */
+function ErrorCard(props: {error: unknown; onRetry: () => void}) {
```

Comment:
The unknown type here and the `stack` and `message` params become a little confusing for me here, but I understand if it's more difficult to more strictly type things here.

Reply:
It follows the precedence in `StudioErrorBoundary`, as well as the convention of the stricter TS rule that enforces:
```ts
try {
  doSomething()
} catch(error: unknown) {
 // `error` is never guaranteed to be a subclass of `Error` and can be anything, `unknown` forces you to deal with every eventuality
}
```

Which is also why the `isRecord` asserters are used further down

Reply:
ahh, of course! apologies for not checking that